packages/js-sdk/src/sandbox/sandboxApi.ts

```diff
@@ -133,8 +168,42 @@ export class SandboxApi {
         ...(sandbox.alias && { name: sandbox.alias }),
         metadata: sandbox.metadata ?? {},
         startedAt: new Date(sandbox.startedAt),
-      })) ?? []
+        state: sandbox.state,
+      })
     )
+
+    return {
+      sandboxes,
+      hasMoreItems,
+      nextToken: nextPageToken,
+      iterator: this.listIterator({
+        limit,
+        nextToken: nextPageToken,
+        query,
+        requestTimeoutMs,
+      }),
+    }
+  }
+
+  private static async *listIterator(
+    options: SandboxListOpts = {}
+  ): AsyncGenerator<SandboxInfo> {
+    let nextPage = true
```

Comment:
what if the list is empty? then `nextPage` should be `false`.

Reply:
tbh. I am still unsure about the iterator here, it feels like now we have two different ways of doing pagination, so I am considering either all-in on iterator or remove it, else it feels half-baked